### PR TITLE
feat(vg_lite): add porting layer for VG-Lite simulator based on ThorVG

### PR DESCRIPTION
**Pull request by XuNeo**
Help us review this PR! Anyone can [approve it or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### Description of the feature or fix

Allow to build vglite simulator using ThorVG.
@FASTSHIFT Please upload your work to `lvgl/src/dev/`.
Locally tested and works.

### Checkpoints
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html)

### CODE REVIEW

File: lv_conf_template.h [hunk 1]
```diff
@@ -140,6 +140,10 @@
 
 /* Enable VG-Lite assert. */
 #define LV_VG_LITE_USE_ASSERT 0
+
+/* Simulate VG-Lite hardware using ThorVG */
+#define LV_USE_VG_LITE_TVG  0
```
Comment (kisvegabor) [inline lv_conf_template.h]:
```suggestion
#define LV_USE_VG_LITE_THORVG  0
```
as we are using the thorvg name on other places too.

**Review by kisvegabor — COMMENTED** [review]
Does it mean we will be able to test the VG-Lite draw_unit APIs and logic in CI?

**Comment by FASTSHIFT** [comment]
> Does it mean we will be able to test the VG-Lite draw_unit APIs and logic in CI?

Exactly!

**Comment by FASTSHIFT** [comment]
At present, there is still a problem that thorvg will automatically clear the buffer. The issue is here: https://github.com/thorvg/thorvg/issues/1901.

File: src/libs/thorvg/tvgSwRenderer.cpp [hunk 2]
```diff
@@ -446,7 +446,11 @@ bool SwRenderer::target(pixel_t* data, uint32_t stride, uint32_t w, uint32_t h,
 
 bool SwRenderer::preRender()
 {
+#if LV_USE_VG_LITE_THORVG
```
Comment (kisvegabor) [inline src/libs/thorvg/tvgSwRenderer.cpp]: Why is it not defined? `lv_conf_internal.h` is included at the top of the file.

Comment (XuNeo) [reply src/libs/thorvg/tvgSwRenderer.cpp]: It's due to the issue mentioned above. To simulate VG-Lite, this function should be bypassed.

> At present, there is still a problem that thorvg will automatically clear the buffer. The issue is here: https://github.com/thorvg/thorvg/issues/1901.

Comment (kisvegabor) [reply src/libs/thorvg/tvgSwRenderer.cpp]: I mean I see this warning:
image

The CI passes, so it should be good, I just wonder why we have this warning.

Comment (XuNeo) [reply src/libs/thorvg/tvgSwRenderer.cpp]: Ah, get it. It should be fixed now. Let wait for CI.

**Review by kisvegabor — APPROVED** [review]

**Comment by kisvegabor** [comment]
Something broke the docs build: https://github.com/lvgl/lvgl/actions/runs/7474817972

**Pull request merged**